repm/api.go

```diff
@@ -357,10 +357,17 @@ func (conn *connection) dispatchCommitTransaction(reqBytes []byte) ([]byte, erro
 	}
 	conn.removeTransaction(req.TransactionID)
 	commitRef, err := tx.Commit()
+
+	res := commitTransactionResponse{}
+
 	if err != nil {
-		return nil, err
+		var commitErr db.CommitError
+		if !errors.As(err, &commitErr) {
+			return nil, err
+		}
+		res.RetryCommit = true
```

Comment:
isn't it really retry*transaction* that we are trying to say here? like you can't just retry the tx.commit() and think it is going to work, the whole transaction seems like it needs to be rerun on the new master. (unless i'm misunderstanding something.)

Reply:
Yup. The whole transaction should be retried. Will rename later.